File: plugins/ruby-kit/testdata/good/app/ok.rb
# Good fixture: binds, allowlisted dispatch, documented allow.

def find_ok(id)
  User.where(id: id).first
end

def find_bind_ok(name)
  User.where("name = ?", name).first
end

ALLOWED = %w[draft publish archive].freeze

def dispatch_ok(action, record)
  raise ArgumentError, "unknown action" unless ALLOWED.include?(action)
  record.public_send(action)
end

# Documented intentional seam (boot probe); keep allow on the smell line.
def documented_legacy(code)
  eval(code) # ruby-rg-allow: fixture documents allow marker for intentional eval seam
end

File: plugins/zig-kit/testdata/good/src/ok.zig
// Good fixture: try/errdefer, documented allow.
const std = @import("std");

pub fn copyOk(allocator: std.mem.Allocator, name: []const u8) ![]u8 {
    return try allocator.dupe(u8, name);
}

// Documented intentional seam (boot probe); keep allow on the smell line.
pub fn documentedLegacy(allocator: std.mem.Allocator) []u8 {
    return allocator.alloc(u8, 8) catch unreachable; // zig-rg-allow: fixture documents allow marker for intentional unchecked seam
}
